.github/workflows/auto-merge.yml

```diff
@@ -0,0 +1,66 @@
+name: Merge head into child branch
+
+on:
+  workflow_dispatch:
+    inputs:
+      head:
+        description: 'Branch to merge from'
+        required: true
+        type: choice
+        options:
+          - stable
+          - minor-next
+      base:
+        description: 'Branch to merge into'
+        required: true
+        type: choice
+        options:
+          - minor-next
+          - major-next
```

Comment:
These inputs are unnecessary. The workflow needs only to do this:

```
git checkout minor-next
git merge stable
git checkout major-next
git merge minor-next
git push origin minor-next major-next
```

The rest is just unnecessary inconvenience.

Reply:
I personally consider that the current form is more beneficial, if a specific merger causes conflicts, it can be handled and resolved in isolation, without interrupting or complicating other mergers.

In the proposed implementation an error in one merger will block the whole process.

Reply:
If a conflict arises, you'll have to re-run a workflow anyway. The only thing this achieves is making it more inconvenient to do so, while creating extra merges between outdated branches.

In any case, a matrix for this is an unnecessary amount of clicks. At most it needs 2 options:
- stable -> minor-next
- minor-next -> major-next